src/GafferImage/FilterAlgo.cpp

```diff
@@ -0,0 +1,222 @@
+	// and we can only do that by looping over the table of registered
+	// filters.
+	for( int i = 0, e = OIIO::Filter2D::num_filters();  i < e;  ++i )
+	{
+		OIIO::FilterDesc fd;
+		OIIO::Filter2D::get_filterdesc( i, &fd );
+		if( !strcmp( fd.name, filterName ) )
+		{
+			return Filter2DPtr(
+				OIIO::Filter2D::create( filterName, fd.width, fd.width ),
+				OIIO::Filter2D::destroy
+			);
+		}
+	}
+
+	if( name == "smoothGaussian" )
+	{
+		/// \todo Perhaps we could add a registration mechanism to OIIO so we could register
+		/// our filter and look it up using the mechanism above?
+		return Filter2DPtr( new SmoothGaussian2D( 3, 3 ) );
+	}
+
+	throw IECore::Exception( boost::str( boost::format( "Unknown filter \"%s\"" ) % filterName ) );	
+}
+
+float GafferImage::FilterAlgo::sample( Sampler &sampler, const V2f &p, const V2f &dpdx, const V2f &dpdy, const OIIO::Filter2D *filter )
+{
+
+	// Multiplier to make sure our filter size never goes less than 1 pixel
+	// \todo : It could still effectively go less than 1 pixel on one axis due
+	// to the two axes being aligned.  Contemplate if this is a real problem
+	float minimumLenMultDx = 1;
+	float minimumLenMultDy = 1;
+	if( dpdx.length2() < 1.0f )
+	{
+		minimumLenMultDx = 1.0 / dpdx.length();
+	}
+	if( dpdy.length2() < 1.0f )
+	{
+		minimumLenMultDy = 1.0 / dpdy.length();
+	}
+
+	// Vectors for half of our current dPdx or dPdy
+	// These get us from the center of the parallelogram for this sample to the corner
+	V2f hdpdx = 0.5f * minimumLenMultDx * dpdx * filter->width();
+	V2f hdpdy = 0.5f * minimumLenMultDy * dpdy * filter->height();
+
+
+	// Invert the 2x2 Matrix formed by the derivative vectors so we can go
+	// from source pixels into the filter's coordinate system
+	float determinant = hdpdx[0] * hdpdy[1] - hdpdx[1] * hdpdy[0];
+	V2f filterWidthMult( filter->width(), filter->height() );
+	V2f xstep = 0.5f / determinant * V2f( hdpdy[1], -hdpdx[1] ) * filterWidthMult;
+	V2f ystep = 0.5f / determinant * V2f( -hdpdy[0], hdpdx[0] ) * filterWidthMult;
+	
+
+	// Find bounds of the 4 corners of our filter spot
+	int minX = INT_MAX;
+	int maxX = INT_MIN;
+	int minY = INT_MAX;
+	int maxY = INT_MIN;
+	for( int corner = 0; corner < 4; corner++ )
+	{
+		V2f cp = p + (( corner & 1 ) ? 1.0f : -1.0f ) * hdpdx + (( corner & 2 ) ? 1.0f : -1.0f ) * hdpdy;
+
+		maxX = std::max( maxX, (int)floorf( cp[0] ) );
+		maxY = std::max( maxY, (int)floorf( cp[1] ) );
+		minX = std::min( minX, (int)ceilf( cp[0] ) );
+		minY = std::min( minY, (int)ceilf( cp[1] ) );
+	}
+
+	float totalW = 0.0f;
+	float v = 0.0f;
+	for( int y = minY; y <= maxY; y++ )
+	{
+		for( int x = minX; x <= maxX; x++ )
+		{
+
+			// \todo : I want to remove some pointless multiplies from here
+			V2f offset = V2f( x, y ) - p;
+			V2f filterPos = offset[0] * xstep + offset[1] * ystep;
+			float w = (*filter)( filterPos[0], filterPos[1] );
+
+			// \todo : I can't think of any way to keep this around cleanly for testing, since
```

Comment:
Is this pretty much equivalent to a disk filter? Could you get the same effect by passing a disk filter in from the outside? Or is it important to be `0.1f` rather than `0.0f` outside the filter radius?